File: src/main/java/com/pcb/pcbridge/players/PlayerData.java
package com.pcb.pcbridge.players;

public final class PlayerData 
{
	public long UserId;
	public boolean SwearFilter;
	
	public PlayerData() { }
	public PlayerData(long userId, boolean swearFilter)
	{
		this.UserId = userId;
		this.SwearFilter = swearFilter;
	}
}

File: pcbridge-http/src/main/kotlin/com/projectcitybuild/pcbridge/http/pcb/models/PlayerData.kt
package com.projectcitybuild.pcbridge.http.pcb.models

import com.google.gson.annotations.SerializedName
import kotlinx.serialization.Serializable

@Serializable
data class PlayerData(
    @SerializedName("account")
    val account: Account? = null,

    @SerializedName("player")
    val player: Player? = null,

    @SerializedName("roles")
    val roles: List<Role> = emptyList(),

    @SerializedName("badges")
    val badges: List<Badge> = emptyList(),

    @SerializedName("elevation")
    val elevation: HttpOpElevation? = null,
) {
    val isStaff: Boolean
        get() = roles.firstOrNull { it.roleType?.lowercase() == "staff" } != null
}
